pkg/controller/jobframework/reconciler.go

```diff
@@ -469,16 +472,18 @@ func (r *JobReconciler) IsParentJobManaged(ctx context.Context, jobObj client.Ob
 }
 
 func (r *JobReconciler) getParentWorkload(ctx context.Context, job GenericJob, object client.Object) (*kueue.Workload, error) {
-	pw := kueue.Workload{}
-	namespacedName := types.NamespacedName{
-		Name:      ParentWorkloadName(job),
-		Namespace: object.GetNamespace(),
+	owner := metav1.GetControllerOf(object)
+	if owner == nil {
+		return nil, nil
 	}
-	if err := r.client.Get(ctx, namespacedName, &pw); err != nil {
+	wlList := kueue.WorkloadList{}
+	if err := r.client.List(ctx, &wlList, client.InNamespace(object.GetNamespace()), client.MatchingFields{indexer.OwnerReferenceUID: string(owner.UID)}); err != nil || len(wlList.Items) == 0 {
 		return nil, client.IgnoreNotFound(err)
-	} else {
-		return &pw, nil
 	}
+
+	// In theory the parent can own multiple Workloads, we cannot do too much about it, maybe log it.
+	return &wlList.Items[0], nil
```

Comment:
What if it does not own any workloads, we could fail then. I see this should not happen normally, but may happen is someone removes the workload from parent job manually, probably we don't want risk panicing here.

Reply:
I guess that there isn't such a failure here. Because if the length of the `wlList` is zero, we return nil in Line 481.